Intermittent AttachmentsPullTest failure

This looks different from the issues previously seen in #337, but may be related. The "Socket closed" looks like we're trying to use something already closed (internally I think we would get a broken pipe if it was closed at the server).

>com.cloudant.sync.replication.AttachmentsPullTest > dontPullAttachmentAlreadyPulled[0] FAILED
    java.lang.AssertionError: Create/pull error java.lang.RuntimeException:
java.net.SocketException: Socket closed
        at org.junit.Assert.fail(Assert.java:88)
        at com.cloudant.sync.replication.AttachmentsPullTest.dontPullAttachmentAlreadyPulled(AttachmentsPullTest.java:198)

Saw yet another failure mode with this test:

>com.cloudant.sync.replication.AttachmentsPullTest > dontPullAttachmentAlreadyPulled[0] FAILED

>CouchException: error: file_exists, reason: The database could not be created, the file already exists., statusCode: 412, msg: Precondition Failed, cause: java.io.IOException: Server returned HTTP response code: 412 for URL: http://127.0.0.1:5984/attachments_pull_test1476725683802

>Caused by: java.io.IOException: Server returned HTTP response code: 412 for URL: http://127.0.0.1:5984/attachments_pull_test1476725683802

--- a/cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/replication/ReplicationTestBase.java
+++ b/cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/replication/ReplicationTestBase.java
@@ -45,10 +45,12 @@ public abstract class ReplicationTestBase extends CouchTestBase {
 
     protected CouchConfig couchConfig = null;
 
-    private long dbSuffix = System.currentTimeMillis();
+    private long dbSuffix;
 
     @Before
     public void setUp() throws Exception {
+        // Use a unique suffix for each test
+        dbSuffix = System.currentTimeMillis();
         this.createDatastore();
         this.createRemoteDB();
     }
@@ -57,7 +59,7 @@ public abstract class ReplicationTestBase extends CouchTestBase {
     public void tearDown() throws Exception {
         datastore.close();
         TestUtils.deleteDatabaseQuietly(database);
-        cleanUpTempFiles();
+        cleanUpTempFilesAndDeleteRemote();
     }
 
 
@@ -78,7 +80,7 @@ public abstract class ReplicationTestBase extends CouchTestBase {
         couchClient = remoteDb.getCouchClient();
     }
 
-    protected void cleanUpTempFiles() {
+    protected void cleanUpTempFilesAndDeleteRemote() {
         TestUtils.deleteTempTestingDir(datastoreManagerPath);
         CouchClientWrapperDbUtils.deleteDbQuietly(remoteDb);
     }

